pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -37,81 +37,74 @@ const _ = cockpit.gettext;
  * Enables showing shell and ansible script. Shell one is mandatory and ansible one can be omitted.
  *
  */
-export class ModificationsExportDialog extends React.Component {
-    constructor(props) {
-        super(props);
+const ModificationsExportDialog = ({ show, onClose, shell, ansible }) => {
+    const [active_tab, setActiveTab] = React.useState("shell");
+    const [copied, setCopied] = React.useState(false);
+    const [timeoutId, setTimeoutId] = React.useState(null);
```

Comment:
Not a fan. This shouldn't trigger renders. That's not a veto though.

Reply:
Hmm it will trigger a re-render anyway because it sets copied to false? Although it should not if you switch tabs I suppose.

Reply:
Ah yes, good point. I hope react coalesces these into one re-render.

Reply:
It should I think, if you call multiple setState's react does collect them and turns them into one state change.